src/streams.c

```diff
@@ -81,35 +82,56 @@ static Int READ_COMMAND(Obj *evalResult)
 
 
 /*
- * This function reads all code from the stream and returns either `fail` if the stream
- * cannot be opened, or executes all statements in the stream and returns a list of lists,
- * each entry of which reflects the result of the execution of one statement.
+ * This function reads all code from the stream and returns either `fail` if
+ * the stream cannot be opened, or executes all statements in the stream and
+ * returns a list of lists, each entry of which reflects the result of the
+ * execution of one statement.
  *
- * The results are returned as lists with at most three bound positions.
+ * The results are returned as lists of length three if printToOutput is
+ * false, or five, if printToOutput is true.
  *
- * If the first entry is `true`, then the second entry is bound to the result of
- * the statement if there was one, and unbound otherwise, and the third entry
- * reflects whether the statement ended in a dual semicolon,
- * if the first entry is `false', an error occurred during the execution of a statement,
- * and no other positions of the list are bound.
+ * If the first entry is `true`, then the second entry is bound to the result
+ * of the statement if there was one, and unbound otherwise, and the third
+ * entry reflects whether the statement ended in a dual semicolon, if the
+ * first entry is `false', an error occurred during the execution of a
+ * statement, and no other positions of the list are bound. If printToOutput
+ * is true, the fourth entry of the list is ViewObj applied to the result, if
+ * there was one, and if the statement did not end in a dual semicolon.
+ * Furthermore, if printToOutput is true, the fifth entry contains a string of
+ * all output printed while the corresponding statement was executed.
  *
- * This function is currently used in interactive tools such as the GAP Jupyter kernel to
- * execute cells and is likely to be replaced by a function that can read a single command
- * from a stream without losing the rest of its content.
+ * This function is currently used in interactive tools such as the GAP
+ * Jupyter kernel to execute cells and is likely to be replaced by a function
+ * that can read a single command from a stream without losing the rest of its
+ * content.
  */
-Obj FuncREAD_ALL_COMMANDS( Obj self, Obj instream, Obj outstream, Obj echo )
+Obj FuncREAD_ALL_COMMANDS(Obj self, Obj instream, Obj echo, Obj printToOutput)
 {
     ExecStatus status;
     Int resultCount, resultCapacity;
     UInt dualSemicolon;
     Obj result, resultList;
     Obj evalResult;
+    Obj        outstream;
+    Obj        outstreamString;
+    Obj        outstreamStringCopy;
```

Comment:
I think you should clang-format the whole thing to make formatting consistent.

Reply:
Ähm. This was not done by me, but I think by clang-format (did not check its results :/).